tools/src/icon4py/tools/py2fgen/wrappers/diffusion_wrapper.py

```diff
@@ -282,6 +304,14 @@ def diffusion_run(
         rho=rho,
     )
 
+    if hdef_ic is None:
+        hdef_ic = gtx.zeros(w.domain, dtype=w.dtype, allocator=backend)
```

Comment:
side-note: the `backend` here is taken from the global config, but for now I would keep it.